ligo/stablecoin/actions.ligo

```diff
@@ -0,0 +1,288 @@
+    Tezos.transaction (responses, 0mutez, parameter.1)
+; const upd : list (operation) = list [transfer_operation]
+} with (upd, store)
+
+(*
+ * Retreives a list of `total_supply` that is specified for multiple
+ * token types in FA2 spec and leave the storage untouched. In reality
+ * we still restrict all of them to be of `default_token_id`.
+ *)
+function total_supply
+  ( const parameter : total_supply_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ // We suppose that the token is already validated in `get_total_supply`
+  // and equal to `default_token_id` since our contract is monolitic
+  const total_supply_response_ : total_supply_response_ = record
+    [ token_id     = default_token_id
+    ; total_supply = store.total_supply
+    ]
+; const total_supply_response : total_supply_response =
+    Layout.convert_to_right_comb ((total_supply_response_ : total_supply_response_))
+  // We drop `token_id` parameter since our contract is monolitic
+  // and so we support only one `default_token_id` token type
+; function get_total_supply
+    ( const ops      : list (total_supply_response)
+    ; const token_id : token_id
+    ) : list (total_supply_response) is block
+    { validate_token_type (token_id)
+    } with total_supply_response # ops
+} with
+  ( list [Tezos.transaction
+    ( List.fold
+        ( get_total_supply
+        , parameter.0
+        , (nil : list (total_supply_response))
+        )
+    , 0mutez
+    , parameter.1
+    )]
+  , store
+  )
+
+(*
+ * Returns a list of `token_metadata` calls. This is supposed
+ * to be used for multiple token types but in this smart-contract
+ * we restrict all of them to be of `default_token_type`. Fails
+ * if one of the parameters has different token id that is
+ * specified
+ *)
+function token_metadata
+  ( const parameter : token_metadata_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ const token_metadata : token_metadata =
+    Layout.convert_to_right_comb ((store.token_metadata : token_metadata_))
+  // We drop `token_id` parameter since our contract is monolitic
+  // and so we support only one `default_token_id` token type
+; function get_token_metadata
+    ( const ops      : list (token_metadata_response)
+    ; const token_id : token_id
+    ) : list (token_metadata_response) is block
+    { validate_token_type (token_id)
+    } with token_metadata # ops
+} with
+  ( list [Tezos.transaction
+    ( List.fold
+        ( get_token_metadata
+        , parameter.0
+        , (nil : list (token_metadata_response))
+        )
+    , 0mutez
+    , parameter.1
+    )]
+  , store
+  )
+
+(*
+ * Retreive current permissions for stablecoin smart contract
+ *)
+function permission_descriptor
+  ( const parameter : permissions_descriptor_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ const permissions : permissions_descriptor =
+    Layout.convert_to_right_comb ((store.permissions : permissions_descriptor_))
```

Comment:
It might be possible to hard code the permission descriptor here. But not yet I guess.